docs/_design/map-controller.md

```diff
@@ -0,0 +1,399 @@
+kind: MapController
+metadata:
+  name: snapshotschedule-controller
+spec:
+  parentResource:
+    apiVersion: snapshot.k8s.io/v1
+    resource: snapshotschedules
+  inputResources:
+  - apiVersion: v1
+    resource: persistentvolumeclaims
+  outputResources:
+  - apiVersion: volumesnapshot.external-storage.k8s.io/v1
+    resource: volumesnapshots
+  resyncPeriodSeconds: 5
+  hooks:
+    map:
+      webhook:
+        url: http://snapshotschedule-controller.metacontroller/map
+    tombstone:
+      webhook:
+        url: http://snapshotschedule-controller.metacontroller/tombstone
+```
+
+### Parent Resource
+
+The parent resource is the SnapshotSchedule itself, and anything this controller
+creates will by owned by this parent.
+The schedule thus acts like a bucket containing snapshots: if you delete the
+schedule, the snapshots inside it will go away too, unless you use
+`--cascade=false` to orphan them.
+Notably, this ties the lifecycles of snapshots to the reason they exist
+(the backup policy that the user defined), rather than tying them to the entity
+that they are about (the PVC).
+
+### Input Resources
+
+The input resources (in this case just PVC) are the inputs to the conceptual
+"map" function.
+We allow multiple input resources because users might want to write a controller
+that performs the same action for several different input types.
+We shouldn't force them to create multiple MapControllers with largely identical
+behavior.
+
+The duck-typed `spec.selector` field (assumed to be `metav1.LabelSelector`) in
+the parent object is used to filter which input objects to process.
+If the selector is empty, we will process all objects of the input types in the
+same namespace as the parent.
+
+If there are multiple input resources, they are processed independently, with no
+attempt to correlate them.
+That is, the [map hook][] will still be called with only a single input object
+each time, although the kind of that object might be different from one call to
+the next.
+
+[map hook]: #map-hook
+
+### Output Resources
+
+The output resources (in this case just VolumeSnapshot) are the types of objects
+that the user intends to create and hold in the conceptual "bucket" that the
+parent object represents.
+We allow multiple output resources because users might think of their controller
+as spitting out a few different things.
+We shouldn't force them to create a CompositeController too just so they can
+emit multiple outputs, especially if those outputs are not conceptually part of
+one larger whole.
+
+For a given input object, the user can generate any number of output objects.
+We will [tag](#map-key) those output objects in some way to associate them with
+the object that we sent as input.
+The tag makes it possible to group those objects and send them along with future
+[map hook requests](#map-hook-request).
+
+If the associated input object is gone, we will detect that there is a group of
+output objects that have become orphaned.
```

Comment:
Not sure if I understand this part. Who orphaned the output objects?

Reply:
I added some clarification that this doesn't refer to orphaning in the ownerRef/controllerRef sense. I wish there was another word I could use that doesn't have a collision with another common API machinery term, but I can't think of one yet. Any ideas?

Reply:
Will `disconnected` be a good word? Perhaps can set up a new notion of `connection`, which is different from `orphaning` and `adoption`.

Reply:
Another suggestion is the notion of `decay`, which may have a more vivid tone. If an input object is removed, the output objects that are not deleted are said to be `decaying`.

Reply:
I see. I did confuse "orphaning" with ownerRef orphaning. Because parent/child metaphor is used for parent & outputs, the use of the term "orphan" could cause confusion. The clarification helps. 

Proposing some more alternative terms: "lost", "untraceable".